framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/LdapDeploymentImpl.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.cloud.openshift.deployment;
+
+import org.kie.cloud.api.deployment.LdapDeployment;
+import org.kie.cloud.openshift.resource.Project;
+
+public class LdapDeploymentImpl extends OpenShiftDeployment implements LdapDeployment {
+
+    private String serviceName;
+    private String host;
+
+    public LdapDeploymentImpl(Project project) {
+        super(project);
+    }
+
+    @Override
+    public String getServiceName() {
+        if (serviceName == null) {
+            serviceName = ServiceUtil.getLdapServiceName(getOpenShift());
+        }
+        return serviceName;
+    }
+
+    @Override
+    public String getHost() {
+        if (host == null) {
+            host = getLdapRoute(getServiceName()).orElseThrow(() -> new RuntimeException("No LDAP URL is available."));
```

Comment:
You seem to use route hostname to access LDAP. Does it work? I thought that OpenShift route is able to handle just HTTP and HTTPS communication.

Reply:
Yes, this is working fine.

Reply:
This has been reverted out.